src/history/index.js

```diff
@@ -14,10 +24,23 @@ export async function recordVersion(serviceId, documentType, content, snapshotId
     throw new Error(`A snapshot ID is required to ensure data consistency for ${serviceId}'s ${documentType}`);
   }
 
-  return record({
+  let id = '';
+  if (config.get('history.publish')) {
+    id += config.get('history.publicSnapshotsRepository')
+      .replace('git@github.com:', 'https://github.com/')
+      .replace('.git', '/commit/');
+  }
+  id += snapshotId;
+
+  return versionRecorder.record({
     serviceId,
     documentType,
     content,
-    snapshotId
+    messageToAppend: `This version was recorded after filtering snapshot ${id}`
   });
 }
+
+export async function publish() {
+  snapshotRecorder.publish();
+  return versionRecorder.publish();
```

Comment:
If this is all async, then we should return the sum of both promises. Currently, if the snapshot publish fails, no one will know.